.github/workflows/ci.yml

```diff
@@ -373,89 +326,153 @@ jobs:
-            # Succesfully ran composer update and installed everything
-            rm __update_attempt.txt
-            # Useful to see generated composer.json when diagnosing new bugs
-            cat composer.json
-          elif ! [[ $(cat __update_attempt.txt) =~ 'Root composer.json requires silverstripe/installer' ]] || \
-            [[ $INPUTS_COMPOSER_REQUIRE_EXTRA =~ silverstripe/installer ]] || \
-            [[ $MATRIX_COMPOSER_REQUIRE_EXTRA =~ silverstripe/installer ]] || \
-            ! [[ "${{ matrix.installer_version }}" =~ ^[1-9]\.[0-9]+\.x\-dev$ ]]
-          then
-            # Failed to run composer update and will not attempt requiring different versions of silverstripe/installer
-            cat composer.json
-            cat __update_attempt.txt
-            # Using exit code 2 as it's the same code that composer uses when it fails
-            exit 2
-          elif [[ "${{ matrix.installer_version }}" =~ ^([1-9])\.([0-9]+)\.x\-dev$ ]]; then
-            # gha-generate-matrix will not always provide a compatibile version of silverstripe/installer due to
-            # limitations of knowing what versions of silverstripe/installer the pushed code is compatible with
-            # In this scenario we will usually end up with the latest minor x.dev version of silverstripe installer
-            # If it initially failed on composer update, try some earlier versions of silverstripe/installer
-            echo "Could not run composer update with version of silverstripe/installer, attempting earlier versions"
-            MAJOR=${BASH_REMATCH[1]}
-            MINOR=${BASH_REMATCH[2]}
-            SUCCESSFULLY_REQUIRED='false'
-            for MI in $(( MINOR - 1 )) $(( MINOR - 2 )); do
-              if [[ $SUCCESSFULLY_REQUIRED == 'false' ]]; then
-                # Test if $MI is a negative number
-                if [[ $MI =~ - ]]; then
-                  cat composer.json
-                  if [[ -f _require_attempt.txt ]]; then
-                    cat __require_attempt.txt
-                  else
-                    cat __update_attempt.txt
-                  fi
-                  echo 'Have run out of silverstripe/installer versions to attempt to require'
-                  exit 2
-                else
-                  VERSION="$MAJOR.$MI.x-dev"
-                  echo "Attempting to require silverstripe/installer $VERSION"
-                  composer require silverstripe/installer:$VERSION 2> __require_attempt.txt || :
-                  if ! [[ $(cat __require_attempt.txt) =~ Problem ]]; then
-                    SUCCESSFULLY_REQUIRED='true'
-                    echo "Succesfully required silverstripe/installer $VERSION"
-                    rm __require_attempt.txt
+          if [[ "${{ inputs.composer_install }}" == "true" ]]; then
+            if ! [[ -f composer.lock ]]; then
+              echo "WARNING: composer_install input is true but there is no composer.lock file"
```

Comment:
This feels like it should be an error that stops the build - the whole point of this is to use composer.lock isn't it?
If we do change this to an error though we also need to update the readme to make it clear that'll happen (composer.lock isn't mentioned there at the moment)

Reply:
Have upgraded to `exit 1`

Reply:
Please also update the README to say that a composer.lock file is required when using this input.

Reply:
Updated